coding_conventions.md

```diff
@@ -0,0 +1,154 @@
+# Coding conventions
+
+The following is a list of conventions we follow when working in the [water-abstraction-system](https://github.com/DEFRA/water-abstraction-system) repo.
+
+We make best efforts to follow them when working in the legacy repos. There are often times it makes more sense to follow the conventions that already exist.
+
+- [Add the .js extension](#add-the-js-extension)
+- [File names](#file-names)
+- [Functions](#functions)
+- [Top of .js files](#top-of-js-files)
+  - [Top of test.js files](#top-of-testjs-files)
+
+## Add the .js extension
+
+When linking to an internal module in our `require()` statements _always_ include the `*.js` extension.
+
+```javascript
+// Good
+const RequestLib = require('../lib/request.lib.js')
+
+// Bad
+const RequestLib = require('../lib/request.lib')
+```
+
+We _know_ 😝 that you don't have to and that generally the convention is not to. But we still hold out hope that one day we'll [switch our main repo from CommonJS to ES6 modules](https://github.com/DEFRA/water-abstraction-system). The extension _is_ required when using `import` and the work of migrating is made easier if the existing `require()` statements already include it.
+
+## File names
+
+All file names should be [kebab-case](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/#kebab-case), for example `awesome-people.service.js`
+
+Other naming conventions are
+
+- controllers `thing.controller.js`
+- general modules `thing.lib.js`
+- models `thing.model.js`
+- plugins `thing.plugin.js`
+- presenters `thing.presenter.js`
+- routes `thing.routes.js`
+- services `thing.service.js`
+
+Unit test files should be the same as the thing being tested with `.test` added to the end, for example `thing.service.test.js`. The `test/` folder structure should mirror the `app/` folder.
+
+## Functions
+
+We define our functions as
+
+```javascript
+function helloWorld () {
+  return 'Hello, world!'
+}
+
+// DON'T do this
+const helloWorld = () => {
+  return 'Hello, world!'
+}
+```
+
+When we do need to use arrow functions, parameters are always wrapped in brackets even if there is only one. Also, use _block body_ over _concise body_.
+
+```javascript
+const materials = [
+  'Hydrogen',
+  'Helium',
+  'Lithium',
+  'Beryllium'
+]
+
+// Use Block body version
+materials.forEach((material) => {
+  console.log(`${material} - ${material.length}`)
+})
+
+// Do not use Concise body version
+materials.forEach((material) => console.log(`${material} - ${material.length}`))
+```
+
+## Top of .js files
+
+The top of all `*.js` files should follow this pattern
+
+- `use strict` as first statement
+- JSDoc module declaration (if a `module` or `class`)
+- `require()` statements
+
+The `require()` statements should be grouped by
+
+- external packages
+- internal modules
+- config
+- constants
+- module scoped variables
+
+Place a new line between each one and list entries alphabetically.
```

Comment:
I don't know if this is something we want to make a rule about but I would tend to put destructured functions in alphabetical order too, eg:
```js
const { functionTheFirst, nextFunction, otherFunction } = require('./coolStuff.service.js')
```

The exception would be Lab imports where I tend to write them in an order that seems logical ie. `describe, it, before, beforeEach, after, afterEach` -- having `before` and `after` in that order makes more sense to me than alphabetically!

Reply:
💯 agree with that take and that it should be included in this initial version. I'll make the change.